packages/logging/src/logging-and-telemetry.ts

```diff
@@ -119,35 +115,43 @@ export class MongoshLoggingAndTelemetry {
     hookLogger(this.bus, this.log, 'mongosh', (uri) =>
       redactURICredentials(uri)
     );
+
+    this.bus.emit('mongosh:logger-initialized');
   }
 
-  private _setupBusEventListeners() {
-    this.bus.on(
-      'mongosh:logger-initialized',
-      this.onLoggerInitialized.bind(this)
-    );
+  private _getTelemetryUserIdentity(): MongoshAnalyticsIdentity {
+    return {
+      anonymousId: this.telemetryAnonymousId ?? (this.userId as string),
+    };
+  }
+
+  /** We emit different analytics events for loading files and evaluating scripts
+   * depending on whether we're already in the REPL or not yet. We store the
+   * state here so that the places where the events are emitted don't have to
+   * be aware of this distinction. */
+  private hasStartedMongoshRepl = false;
+
+  private apiCallTracking = {
```

Comment:
Seemed best to move all variables into the class state and keep the setup listeners function scope clean of state but if it's making the class too bloated, could reconsider (or maybe scope them under `eventListenersState` or something)